routes/controller/friend.js

```diff
@@ -2,22 +2,149 @@ const createError = require("http-errors");
 const mongoose = require("mongoose");
 
 const User = require("../../models/User");
+const Request = require("../../models/Request");
+const defaultOption = require("../../config/paginateOption");
 const ACCESS_LEVELS = require("../../constants/accessLevels");
 const { ERROR } = require("../../constants/messages");
 const {
-  OK, NOT_FOUND, UNAUTHORIZED, BAD_REQUEST,
+  OK, NOT_FOUND, UNAUTHORIZED, BAD_REQUEST, INTERNAL_SERVER_ERROR,
 } = require("../../constants/statusCodes");
 
 async function getFriends(req, res, next) {
-  //
+  try {
+    if (req.accessLevel !== ACCESS_LEVELS.CREATOR) {
+      throw createError(UNAUTHORIZED);
+    }
+
+    const { creator } = mongoose.Types.ObjectId(req);
+    const { page } = req.headers;
+
+    const paginateOptions = {
+      sort: { lastAccessDate: -1 },
+      limit: 8,
+      populate: "friends",
+      lean: true,
+    };
+
+    if (page) {
+      paginateOptions.page = page;
+    }
+
+    const result = await User.paginate({
+      _id: creator,
+    },
+    paginateOptions).exec();
+
+    const receivedRequests = await Request.find({
+      receiverId: creator,
+    });
+
+    const sentRequests = await Request.find({
+      senderId: creator,
+    });
+
+    res.status(OK);
+    res.json({
+      result: "ok",
+      data: result.docs,
+      nextPage: result.nextPage,
+      prevPage: result.prevPage,
+      receivedRequests,
+      sentRequests,
+    });
+  } catch (err) {
+    next();
+  }
 }
 
 async function sendFriendRequest(req, res, next) {
-  //
+  try {
+    const { creator } = req;
+    const { friendId } = req.body;
+
+    if (req.accessLevel !== ACCESS_LEVELS.CREATOR) {
+      throw createError(UNAUTHORIZED);
+    }
+
+    if (!mongoose.Types.ObjectId.isValid(friendId)) {
+      throw createError(BAD_REQUEST, ERROR.INVALID_FRIEND_ID);
+    }
+
+    const prevRequest = await Request.findOne({
+      receiverId: mongoose.Types.ObjectId(friendId),
+      senderId: mongoose.Types.ObjectId(creator),
```

Comment:
creator가 객체로 타입이 바뀌었습니다.
```suggestion
      senderId: mongoose.Types.ObjectId(creator.id),
```

Reply:
객체 타입 변경이 반영되지 않았네요. 수정하겠습니다!